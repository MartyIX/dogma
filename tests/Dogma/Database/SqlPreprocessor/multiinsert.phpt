<?php

/**
 * Test: Nette\Database\SqlPreprocessor: multiinsert
 *
 * @author     Vlasta Neubauer
 * @package    Nette\Database
 * @subpackage UnitTests
 */

require_once __DIR__ . '/../connect.inc.php';


$processor = $connection->getSqlPreprocessor();


Assert::same(array("INSERT INTO table (col1, col2) VALUES (123, 456), (456, 789)", array()), 
	$processor->process("INSERT INTO table (col1, col2) VALUES ?", 
		array(
			array(
				array(123, 456),
				array(456, 789),
			)
		)
	)
);

Assert::same(array("INSERT INTO table (`col1`, `col2`) VALUES (123, 456), (456, 789)", array()), 
	$processor->process("INSERT INTO table ?", 
		array(
			array(
				array('col1' => 123, 'col2' => 456),
				array('col1' => 456, 'col2' => 789),
			)
		)
	)
);

Assert::same(array("INSERT INTO table (col1, col2) VALUES (123, 456), (456, 789)", array()), 
	$processor->process("INSERT INTO table (col1, col2) VALUES ", 
		array(
			array(
				array(123, 456),
				array(456, 789),
			)
		)
	)
);

Assert::same(array("INSERT INTO table (`col1`, `col2`) VALUES (123, 456), (456, 789)", array()), 
	$processor->process("INSERT INTO table ", 
		array(
			array(
				array('col1' => 123, 'col2' => 456),
				array('col1' => 456, 'col2' => 789),
			)
		)
	)
);
