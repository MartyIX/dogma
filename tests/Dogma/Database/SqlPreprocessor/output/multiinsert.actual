array(
	'INSERT INTO table (col1, col2) VALUES (`0`, `1`) VALUES (123, 456), (456, 789)',
	array(),
)