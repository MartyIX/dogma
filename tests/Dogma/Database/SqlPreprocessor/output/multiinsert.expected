array(
	'INSERT INTO table (col1, col2) VALUES (123, 456), (456, 789)',
	array(),
)